package/MDAnalysis/analysis/utils/autocorrelation.py

```diff
@@ -0,0 +1,155 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4
+#
+# MDAnalysis --- https://www.mdanalysis.org
+# Copyright (c) 2006-2017 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+# doi: 10.25080/majora-629e541a-00e
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+
+from __future__ import absolute_import, division
+import numpy as np
+from copy import deepcopy
+
+
+def autocorrelation(list_of_sets, tau_max, window_step=1):
+    r"""The descrete implementation of the autocorrelation function.
+
+    .. math::
+       C_{HB}^c(\tau) = \frac{\sum_{ij}h_{ij}(t_0)h'_{ij}(t_0+\tau)}{\sum_{ij}h_{ij}(t_0)}
+
+    Parameters
+    ----------
+    list_of_sets : list
+      List of sets,
```

Comment:
```suggestion
      List of sets
```

Reply:
What is this?

Reply:
The list of sets  is a general input that allows to use the same autocorrelation function in both cases, the analysis of atom survival probability, as well as hydrogen bond survival. In the case of atom survival, it is a set that contains IDs of the atoms. In the case of hydrogen bond, the set should actually be a tuple of two atom IDs.